include/IECore/Object.h

```diff
@@ -262,6 +262,8 @@ class IECORE_API Object : public RunTimeTyped
 				/// @param throwIfMissing If false will and the container does not carry the entry for the type name, returns a null pointer.
 				ConstIndexedIOPtr container( const std::string &typeName, unsigned int &ioVersion, bool throwIfMissing = true );
 				template<class T>
+
+				/// TODO TODO TODO
```

Comment:
???

Reply:
I've now added a new commit where I add actual cancellation support to Object::load, putting the Canceller in the LoadContext as discussed.

The only tricky part is testing it ... the final commit I've put up hacks support for cancellation into SceneCache in a way that is quite ugly and extremely non-threadsafe ... but it does prove that Object::load is working correctly ( it felt easier than trying to add a test specifically for Object::load with the usual file mechanisms? ).  Is this good enough to know that it's working for now?  I can just rip out the hack and make the test an expectedFailure before we merge?

Reply:
> it felt easier than trying to add a test specifically for Object::load with the usual file mechanisms?

I don't see how this is easier, and in any case it moves the testing further away from the mechanism we're trying to test. Can't we just write a Primitive into a FileIndexedIO or MemoryIndexedIO directly and then read from it with cancellation?